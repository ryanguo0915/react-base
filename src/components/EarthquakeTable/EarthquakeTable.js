import React  from "react";
import { Link } from "react-router-dom";

export default function EarthquakeTable({features}) {

  return (
      <table>
          <thead>
              <tr>
                  <th>
                      Title
                  </th>
                  <th>
                      Magnitute
                  </th>
                  <th>
                      Time
                  </th>
              </tr>
          </thead>
          <tbody>
            {features.map(feature => {
                const date = new Date(feature.properties.time).toLocaleString();
                return (
                    <tr key={feature.id}>
                        <td><Link to={`/earthquake/${feature.id}`}>{feature.properties.place}</Link></td>
                        <td>{feature.properties.mag}</td>
                        <td>{date}</td>
                    </tr>
                );
            })}
          </tbody>
      </table>
  );
}