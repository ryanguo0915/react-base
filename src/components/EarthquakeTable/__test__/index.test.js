import React from 'react';
import { shallow } from 'enzyme';
import EarthquakeTable from '../EarthquakeTable';
describe("Earthquake table", () => {
  it("should render my component", () => {
    const props = { 
        features: [{
            id:'1', 
            properties:{title:'title', mag: 'mag', time: 'time', status: 'status', tsunami: 'tsunami', type: 'type'}
        }], 
    };
    const wrapper = shallow(<EarthquakeTable {...props}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  it("should have 1 matched perperty and render tables", () => {
    const props = { 
        features: [{id:'1', properties:{title:'title', mag: 'mag', time: 'time', status: 'status', tsunami: 'tsunami', type: 'type'}}], 
    };
    const wrapper = shallow(<EarthquakeTable {...props}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
    const table = wrapper.find('table');
    const row = wrapper.find('tr');
    expect(table).toHaveLength(1);
    expect(row).toHaveLength(2);
  });
});