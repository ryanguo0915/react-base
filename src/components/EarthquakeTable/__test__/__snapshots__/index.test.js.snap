// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Earthquake table should have 1 matched perperty and render tables 1`] = `
Array [
  <table>
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Magnitute
        </th>
        <th>
          Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <Link
            to="/earthquake/1"
          />
        </td>
        <td>
          mag
        </td>
        <td>
          Invalid Date
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`Earthquake table should render my component 1`] = `
Array [
  <table>
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Magnitute
        </th>
        <th>
          Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <Link
            to="/earthquake/1"
          />
        </td>
        <td>
          mag
        </td>
        <td>
          Invalid Date
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
